src/ansys/aedt/core/visualization/advanced/touchstone_parser.py

```diff
@@ -111,6 +112,85 @@ def __init__(self, solution_data=None, touchstone_file=None):
             rf.Network.__init__(self, touchstone_file)
         self.log_x = True
 
+    @pyaedt_function_handler()
+    def get_coupling_in_range(
+        self, start_frequency=1e9, low_loss=-40, high_loss=-60, frequency_sample=5, output_file=None
+    ):
+        """Get coupling losses, excluding return loss, that has at least one frequency point between a range of
+        losses.
+
+        Parameters
+        ----------
+        start_frequency : float, optional
+            Specify frequency value below which not check will be done. The default is ``1e9``.
+        low_loss: float, optional
+            Specify range lower loss. The default is ``-40``.
+        high_loss: float, optional
+            Specify range higher loss. The default is ``-60``.
+        frequency_sample : integer, optional
+            Specify frequency sample at which coupling check will be done. The default is ``5``.
+        output_file : path, optional
+            Output file path to save where identified coupling will be listed. The default is ``None``.
+
+        Returns
+        -------
+         list
+            List of S parameters in the range [high_loss, low_loss] to plot.
+
+        """
+
+        nb_freq = self.frequency.npoints
+        k = 0
+        k_start = 0
+
+        # identify frequency index at which to start the check
+        while k < nb_freq:
+            if self.frequency.f[k] >= start_frequency:
+                k_start = k
+                break
+            else:
+                k = k + 1
+
+        port_names = self.port_names
+        nb_port = len(port_names)
+        s_db = self.s_db[:, :, :]
+        temp_list = []
+        temp_file = []
+
+        i = 0
+        j = 0
+
+        while i < nb_port:
+            while j < nb_port:
+                if i != j:
+                    k = k_start
+                    while k < nb_freq:
+                        loss = s_db[k, i, j]
+                        if high_loss < loss < low_loss:
+                            temp_list.append((i, j))
+                            sxy = "S(" + self.port_names[i] + " , " + self.port_names[j] + ")"
+                            str_loss = " Loss = {:.2f} dB "
+                            str_loss_f = str_loss.format(loss)
+                            freq = " Freq = {:.3f} GHz"
+                            freq_f = freq.format((self.f[k]) * 1e-9)
+                            temp_file.append(sxy + str_loss_f + freq_f + "\n")
+                            k = k + frequency_sample
+                            break
+                        else:
+                            k = k + frequency_sample
+                            continue
+                j = j + 1
+            i = i + 1
+            j = i
```

Comment:
```suggestion
for i in range(nb_port):
    for j in range(nb_port):
        if i == j:
            continue
        for k in range(k_start, nb_freq, frequency_sample):
            loss = s_db[k, i, j]
            if high_loss < loss < low_loss:
                temp_list.append((i, j))
                sxy = f"S({self.port_names[i]} , {self.port_names[j]})"
                line = f"{sxy} Loss = {loss:.2f} dB Freq = {(self.f[k] * 1e-9):.3f} GHz\n"
                temp_file.append(line)
                break
```
Btw, I was confused by `low_loss` and `high_loss` values at first. I had to do some research to understand it xD

Reply:
Loss is negative in dB so as an example low_loss is -20dB as high_loss is -40dB.

Reply:
HI @SMoraisAnsys 

I check the results comparing your suggestion to the original code for the log file and the results is different so I've to check why. Your suggestion identify more candidate for the test "if high_loss < loss < low_loss:"

I would commit your change as it is much cleaner but I need to understand before what is the difference.

Reply:
Hi @SMoraisAnsys 

I know why we have a difference because of reciprocity I check only 1 side of the matrix:
![image](https://github.com/user-attachments/assets/bc43c074-a97b-4d5f-9632-d31f2e390f02)
Your suggested code check the two side of the matrix
such you can see below that in the log file you have both S(MA4_I , MA4_O) and S(MA4_O , MA4_I)
![image](https://github.com/user-attachments/assets/de07abdf-d37c-4262-bbba-a6614da7155b)

I think in my code I achieve it by doing 
j = i
at the end of the while j < nb_port: loop

Do you have a suggestion to fix this?